server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +267,68 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            if (!isSearchable()) {
+                return SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0);
+            }
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            return query;
+        }
+
+        @Override
+        public Query termsQuery(List<?> values, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            if (!isSearchable()) {
+                long[] v = new long[values.size()];
+                for (int i = 0; i < v.length; i++) {
+                    v[i] = Values.TRUE.bytesEquals(indexedValueForSearch(values.get(i))) ? 1 : 0;
+                }
```

Comment:
There are only 2 possible values here, right? If yes, we should avoid allocating the array of more than 2 length. Also, isn't it `field_exists` query if both true/false values are present, since it is boolean field?

Reply:
Right -- there are essentially 3 cases for term-in-set, range, and wildcard query:

1. Both true and false values are present -> This is a `field_exists` query.
2. Neither true nor false values are present -> This is a `match_none` query.
3. Exactly one of true or false is present -> delegate to `termQuery`.

I would suggest range + wildcard delegate to `termsQuery`.

Reply:
@jainankitk @msfroh I went back to the older approach of delegating to `termQuery`. I think the mistake I made previously was to use a `MatchAllDocs` instead of a `FieldExists`.